hibernate-core/src/main/java/org/hibernate/dialect/DerbyDialect.java

```diff
@@ -87,11 +91,14 @@ public DerbyDialect(int version) {
 		registerColumnType( Types.BIT, 1, "boolean" ); //no bit
 		registerColumnType( Types.BIT, "smallint" ); //no bit
 		registerColumnType( Types.TINYINT, "smallint" ); //no tinyint
+		registerColumnType( Types.CHAR, "char(1)" );
```

Comment:
This should not be necessary, and is probably wrong.

Reply:
Adapted from https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/dialect/DB2Dialect.java#L95 which was the super type of the derby dialect

Reply:
Yeah but if you look in `Dialect`, we now have `registerColumnType( Types.CHAR, "char($l)" );`, and you're overriding that here with `"char(1)"` which wrongly treats the SQL `char()` type as a single character instead of a string.